egs/cfg2/wip/kaldi_grammar.py

```diff
@@ -0,0 +1,121 @@
+#! /usr/bin/env python
+import os
+from grammar_parser.cfgparser import CFGParser
+
+
+class KaldiGrammar:
+    """
+    Class KaldiGrammar uses as input a grammar file with extension '.fcfg' and has two functions:
+    get_rule_element: extracts the defined grammar rules
+    get_words: extracts the unique words and creates 'corpus.txt' which is used to build 'G.fst'
+    """
+    def __init__(self, grammar_file_string, target):
+
+        if os.path.exists(grammar_file_string):
+            self.parser = CFGParser.fromfile(grammar_file_string)
+            self.grammar_file = grammar_file_string
+        else:
+            self.parser = CFGParser.fromstring(grammar_file_string)
+            self.grammar_string = grammar_file_string
+
+        self.target = target
+
+    def get_rule_element(self, depth=0):
+        """
+        Extracts the grammar rules, defined within the grammar file
+
+        :param depth: level of depth of the grammar rule
+        :return: grammar tree
+        """
+        RulesList = {}
+        # print(depth)
+        if self.target not in self.parser.rules:
+            raise Exception("Target {} not in parser rules".format(self.target))
+
+        # If already present in RULES, return it
+        if self.target in RulesList:
+            return RulesList[self.target]
+
+        # Get the rule
+        rule = self.parser.rules[self.target]
+
+        # Iterate over all options
+        option_alternative_list = []
+        for opt in rule.options:
+
+            # Iterate over all conjunctions
+            conjunctions_list = []
+            for conj in opt.conjuncts:
+                # If the conjunction is already present
+                if conj.name in RulesList:
+                    conjunctions_list.append(RulesList[conj.name])
+                    continue
+
+                # If variable: go one level deeper
+                if conj.is_variable:
+                    result = get_rule_element(conj.name, self.parser, depth + 1)
```

Comment:
1. Add self to the function name -> `self.get_rule_element`
2. Remove self.parser from argument

Reply:
done